use-cases.md

```diff
@@ -69,11 +65,7 @@ The `operation` above refers to the name the server wants to give to the Span. F
 The `span` object above can be `None` if the Tracer did not find relevant headers in the incoming request: presumably because the client did not send them. In this case the server needs to start a brand new trace.
 
 ```python
-    span = tracer.join_trace_from_text(
-        operation_name=operation,
-        context_snapshot=request.headers,
-        trace_attributes=request.headers
-    )
+    span = tracer.get_span_extractor(request.class).extract(request)
```

Comment:
1) That's a possibility. I'm thinking though of using an Inversion of Control here so that Injector/Extractor implementations can claim responsibility of what objects that can handle, rather than the user having to know. A problem with the tracer extract method is it's a plumbing api (convenience method on top of another).

    tracer.extract(object)
    tracer.getSpanExtractor(object.class).extract(object)

2) This can still be done with my original proposal

    SpanExtractor custom = new StringMapExtractor(false);
    Span span = custom.extract(map);
Here the use of explicitly calling constructors intentionally takes us into the land of the internal APIs.

4) Yes there's no reason that Injector and Extractor don't sit in once interface. Just didn't have a good name for it, and wanted a little added clarity at this point.